### PR TITLE
feat!: QC report

### PR DESCRIPTION
**Pull request by bioinformaticsguy**
I have implemented the qc_report rule. 

Once the workflow finishes the qc_report can be generated by calling a similar command. 

snakemake -c 11 --use-conda --snakefile ../../workflow/Snakefile --configfile config.yml --notemp -p qc_report

Do not forget to mention the report rule name qc_report at the end of the snakemake call. 

I have only tested this qc_report on in the count workflow. There are potential bugs that will be fixed in future pull requests.

### CODE REVIEW

**Comment by bioinformaticsguy** [comment]
I have checked it on the combined workflow and the count workflow as well. It generates the report for multiple experiments. Kindly check and merge the pull request. 

P.S: You might find a conflict with the config file, that was accidentally committed at some point. We can totally ignore the conflicts in the config file.

**Comment by visze** [comment]
@ bioinformaticsguy What is the status here? Shoudl I check and merge?

**Comment by bioinformaticsguy** [comment]
It would be great if you can test it locally and merge!

**Review by visze — APPROVED** [review]
LGTM. thx

File: workflow/rules/qc_report.smk [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        quarto_file = temp("results/experiments/{project}/qc_report/qc_report.{assignment}.{config}.{type}.{condition}.{raw_or_assigned}.qmd"),
+    conda:
+        "../envs/quarto.yaml",   
+    params:
+        condition = lambda wildcards: getConditions(wildcards.project),
+        workdir = os.getcwd(),
+
+    shell:
+        """
+        cp config.yml results/experiments/{wildcards.project}/qc_report/config.yml;
```
Comment (visze) [inline workflow/rules/qc_report.smk:79]: I think the config can be removed, right?

File: workflow/rules/qc_report.smk [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        -P rna_oligo_coor_plot:{input.rna_oligo_coor_plot} \
+        -P ratio_oligo_coor_plot:{input.ratio_oligo_coor_plot} \
+        -P ratio_oligo_min_thre_plot:{input.ratio_oligo_min_thre_plot} \
+        -P statistics_all:{input.statistics_all} \
+        -P per_bar_code_dna:{input.per_bar_code_dna} \
+        -P bc_coor_dna:{input.bc_coor_dna} \
+        -P bc_coor_rna:{input.bc_coor_rna} \
+        -P bc_coor_ratio:{input.bc_coor_ratio} \
+        -P workdir:{params.workdir}
+        rm config.yml
```
Comment (visze) [inline workflow/rules/qc_report.smk:97]: same here

**Pull request closed without merging**